pkg/chunkenc/xor.go

```diff
@@ -163,6 +163,11 @@ func (a *xorAppender) Append(t int64, v float64) {
 	var tDelta uint64
 	num := *a.samples
 
+	// Do not append if sample is too old.
+	if t < a.t {
+		return
```

Comment:
please add a debug print for discarding this sample

Reply:
That's actually not possible without turning this PR into a logger refactoring PR.